### PR TITLE
Improved Tests

### PR DESCRIPTION
**Pull request by Tang8330**
Let's merge this before: https://github.com/artie-labs/transfer/pull/385 so we have additional test coverage.

## Changes

1. Adding more tests to `relational_data`
2. Swapped the order of `precision` and `scale` for `NewDecimal` to be consistent with PostgreSQL's NUMERIC data type `NUMERIC(p, s)`
3. Better + consolidated tests for `ParseNumeric`

### CODE REVIEW

File: lib/cdc/util/optional_schema_test.go [hunk 1]
```diff
@@ -1,29 +1,122 @@
...
+
+		actualData := schemaEventPayload.GetOptionalSchema()
+		for actualKey, actualVal := range actualData {
+			testMsg := fmt.Sprintf("key: %s, actualKind: %s, index: %d", actualKey, actualVal.Kind, idx)
+
+			expectedValue, isOk := tc.expected[actualKey]
+			assert.True(t, isOk, testMsg)
+			assert.Equal(t, expectedValue.Kind, actualVal.Kind, testMsg)
+			if expectedValue.ExtendedDecimalDetails != nil || actualVal.ExtendedDecimalDetails != nil {
+				assert.Equal(t, expectedValue.ExtendedDecimalDetails.Scale(), actualVal.ExtendedDecimalDetails.Scale(), testMsg)
```
Comment (nathan-artie) [inline lib/cdc/util/optional_schema_test.go:117]: `assert.NotNil(t, actualVal.ExtendedDecimalDetails, testMsg)`

Comment (nathan-artie) [reply lib/cdc/util/optional_schema_test.go:117]: And then below:
```
} else {
    assert.Nil(t, actualVal.ExtendedDecimalDetails, testMsg)
}```

Comment (nathan-artie) [reply lib/cdc/util/optional_schema_test.go:117]: So that if `ExtendedDecimalDetails` erroneously comes back `nil` for `actualVal` the test fails.

**Review by nathan-artie — APPROVED** [review]

File: lib/cdc/util/optional_schema_test.go (hunk 1 above)
Comment (Tang8330) [reply lib/cdc/util/optional_schema_test.go:118]: I like it

**Pull request merged**